File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/primitive/DoubleFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.primitive;

import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface DoubleFilter {

    AnnotationState status() default AnnotationState.ON;

    double lessThan() default Double.MAX_VALUE;

    double lessOrEqual() default Double.MAX_VALUE;

    double moreThan() default Double.MIN_VALUE;

    double moreOrEqual() default Double.MIN_VALUE;

    double equal() default Double.NaN;

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/primitive/DoubleFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.primitive

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class DoubleFilter(val status: AnnotationState = AnnotationState.ON, val lessThan: Double = java.lang.Double.MAX_VALUE, val lessOrEqual: Double = java.lang.Double.MAX_VALUE, val moreThan: Double = java.lang.Double.MIN_VALUE, val moreOrEqual: Double = java.lang.Double.MIN_VALUE, val equal: Double = java.lang.Double.NaN)
